docs/dev-guide/contributing/branching.rst

```diff
@@ -75,10 +70,9 @@ will be built from the most recent release candidate tag.
 Hotfix
 ------
 
-When a hotfix needs to be made, a branch will be created from the most recent
-``x.y.z`` release tag. The fix will be made (via pull request from a personal
-fork to the hotfix branch), a new tag will be built from the tip of the hotfix
-branch, and the hotfix branch can be merged to ``x.y-dev``.
+When a hotfix needs to be made. The fix will be made (via pull request from a personal
+fork to the ``x.y-release`` branch), a new tag will be built from the tip of the
```

Comment:
We should specify that it should be cherrypicked to master but still merged via a separate PR.

Random sidenote: we need to make a better distinction in our other docs about the difference between a hotfix and a bugfix.  I don't really understand the reasons for the difference in procedure between the two.